sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -52,7 +53,7 @@
 @pytest.mark.parametrize('kernel', kernels)
 def test_kernel_gradient(kernel):
     # Compare analytic and numeric gradient of kernels.
-    K, K_gradient = kernel(X, eval_gradient=True)
+    K, K_gradient = assert_no_warnings(kernel, X, eval_gradient=True)
```

Comment:
I don't think we need to check that no warning is issued. Otherwise we'd have to to that for every line of code. The fact that the warning disappears with this PR should be enough to me.

Reply:
+1 with @jeremiedbb

Reply:
I agree.

Reply:
OK, I agree.

Reply:
I reverted this line to original.